Algorithm.CSharp/Alphas/PumpAndDumpAlpha.cs

```diff
@@ -0,0 +1,129 @@
+
+            // Equally weigh securities in portfolio, based on insights
+            SetPortfolioConstruction(new EqualWeightingPortfolioConstructionModel());
+        }
+
+        /// <summary>
+        /// Performs coarse selection for constituents.
+        /// The stocks must have fundamental data
+        /// The stock must have positive previous-day close price
+        /// The stock must have volume between $1000000 and $10000 on the previous trading day
+        /// The stock must cost less than $5'''
+        /// </summary>
+        private class PennyStockUniverseSelectionModel : FundamentalUniverseSelectionModel
+        {
+            private const int _numberOfSymbolsCoarse = 500;
+            private int _lastMonth = -1;
+
+            public PennyStockUniverseSelectionModel() : base(true, null, null)
+            {
+            }
+
+            public override IEnumerable<Symbol> SelectCoarse(QCAlgorithmFramework algorithm, IEnumerable<CoarseFundamental> coarse)
+            {
+                var month = algorithm.Time.Month;
+                if (month == _lastMonth)
+                {
+                    return algorithm.Universe.Unchanged;
+                }
+                _lastMonth = month;
+
+                return (from cf in coarse
+                        where cf.HasFundamentalData
+                        where cf.Volume < 1000000
+                        where cf.Volume > 10000
+                        where cf.Price < 5
+                        orderby cf.DollarVolume descending
+                        select cf.Symbol).Take(_numberOfSymbolsCoarse);
+            }
+        }
+
+        /// <summary>
+        /// Uses ranking of intraday percentage difference between open price and close price to create magnitude and direction prediction for insights
+        /// </summary>
+        private class PumpAndDumpAlphaModel : AlphaModel
+        {
+            private readonly int _numberOfStocks;
+            private readonly TimeSpan _predictionInterval;
+
+            public PumpAndDumpAlphaModel(
+                int lookback = 1,
+                int numberOfStocks = 10,
+                Resolution resolution = Resolution.Daily)
+            {
+                _numberOfStocks = numberOfStocks;
+                _predictionInterval = resolution.ToTimeSpan().Multiply(lookback);
+            }
+
+            public override IEnumerable<Insight> Update(QCAlgorithmFramework algorithm, Slice data)
+            {
+                return (
+                    from entry in algorithm.ActiveSecurities
+                    let security = entry.Value
+                    where security.HasData && security.Open > 0
+                    // Rank penny stocks on one day price change
+                    let Magnitude = security.Close / security.Open - 1
+                    orderby -Math.Round(Magnitude, 6), security.Symbol
+                    select CreateInsight(security.Symbol, Magnitude))
```

Comment:
Should it verify that the `Magnitude` is `> 0` meaning `security.Close > security.Open`? I think it supposes that there will always be at least `_numberOfStocks` where `Magnitude` is `> 0`

Reply:
We don't want to exclude stocks with negative returns. @HalldorAndersen, can you confirm? Otherwise, we need to update both versions.

Reply:
Correct. We want to include all stocks.

Reply:
Please remove order by negative and use the descending keyword to keep it simple?